build/xwalk.gni

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This file contains Crosswalk-feature-related build flags. Users should
+# import it into their build folder by run:
+#   echo 'import("//xwalk/build/xwalk.gni")' > out/your_folder/args.gn'
+# Then run:
+#   gn gen out/your_folder
+
+# Set it false when building content_shell.
+declare_args() {
```

Comment:
> I understand we should try to not touch the src/BUILD.gn. Actually there is only one flag added in BUILD.gn which is very easy to rebase.

Are you referring to the `building_crosswalk` flag in the chromium-crosswalk pull request? If so, the pull request itself shows that's not the case: at the moment the entire file is being rewritten there because of formatting, and not fixing the formatting means the style checks run in each pull request will always fail.

> I'm not sure if there are more underlying GN target conflicts

This is actually pretty easy to find out: if any conflicts arise `gn gen` will fail to run just like it did now.

> On another hand, To some extent, Crosswalk and Chrome are quite similar apps both based on content_layer [...]. Fix those conflicts will also bring new rebase burden.
> I'm not very sure about this part, please help to give some suggestions, Thanks very much

You are correct that this is different compared to what we currently do with gyp, where `gyp_xwalk` makes gyp start from` xwalk/xwalk.gyp` and ignore the chromium gyp files such as `all.gyp`. That does makes things easier for us, but I don't see an easy way of achieving the same thing with GN, possibly by design: we'd either need to heavily patch `src/BUILD.gn` or add a `.gn` to `src/xwalk` and require people to use `src/xwalk` as the GN root. However, I don't think the latter would even work because if `src/xwalk` it won't be able to reference anything above it in `src`.

So in practice the only ideas I can think of are:
* Patching the Chromium GN code to stop packing the locales (not really feasible unless a flag such as `building_crosswalk` is introduced so not a very good idea)
* `git mv`'ing `src/BUILD.gn` to something else and replacing it with a smaller file that only introduces a flag like `building_crosswalk` and imports either the original `src/BUILD.gn` or `xwalk/BUILD.gn`.
* Not copying the locales to `locales/` and use `xwalk/locales/` instead. The release building code may copy those files manually to `locales`, or our code could be changed to look for the locales in `xwalk/locales` instead.

The 3rd option requires more investigation, but is the cleanest approach. Option 2 is a hack that may be usable for a little while, and option 1 is out of question so far to me.